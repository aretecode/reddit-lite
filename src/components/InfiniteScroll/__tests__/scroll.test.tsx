import * as React from 'react'
import { render } from 'react-testing-library'
import { InfiniteScroll } from '../'
import * as exported from '../'

/**
 * we have it polyfill in jest.setup
 */
describe('InfiniteScroll', () => {
  it('should match export snapshot', () => {
    expect(exported).toMatchSnapshot()
  })
  it('should render without errors', () => {
    const { container } = render(
      <InfiniteScroll
        onReadyToLoadMore={() => {
          throw new Error('never scrolls')
        }}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it.skip('should load more when we scroll', () => {
    // @todo
  })
})
