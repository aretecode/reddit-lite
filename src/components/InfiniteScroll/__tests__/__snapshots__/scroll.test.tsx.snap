// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfiniteScroll should match export snapshot 1`] = `
Object {
  "InfiniteScroll": [Function],
  "StyledInfiniteScrollWrap": Object {
    "$$typeof": Symbol(react.forward_ref),
    "attrs": Array [],
    "componentStyle": ComponentStyle {
      "componentId": "sc-bdVaJa",
      "isStatic": true,
      "rules": Array [
        "
  margin-bottom: 10vh;
  padding-bottom: 1rem;
",
      ],
    },
    "displayName": "styled.div",
    "foldedComponentIds": Array [],
    "render": [Function],
    "styledComponentId": "sc-bdVaJa",
    "target": "div",
    "toString": [Function],
    "warnTooManyClasses": [Function],
    "withComponent": [Function],
  },
  "default": [Function],
}
`;

exports[`InfiniteScroll should render without errors 1`] = `
.c0 {
  margin-bottom: 10vh;
  padding-bottom: 1rem;
}

<div>
  <div
    class="c0"
  />
</div>
`;
